### PR TITLE
Introduce Integration Tests

### PR DESCRIPTION
**Pull request by lucemans**
This PR introduces a `test` folder, included within a set of tests, test runner framework, and data, using bun.

In addition to the above this PR introduces CI/CD pipeline modifications that auto-run tests on PRs and Builds!

As a cherry on top this PR (oopsy) introduces [mozilla/sccache](https://github.com/mozilla/sccache) speeding up ci builds by a ton!

### CODE REVIEW

File: test/data/basic.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const dataset_names_basic: [string, string, Partial<{ address: string }>][] = [
+    ["ETHRegistry_1", "luc.eth", { address: "0x225f137127d9067788314bc7fcc1f36746a3c3B5" }],
+    ["ETHRegistry_2", "nick.eth", { address: "0xb8c2C29ee19D8307cb7255e1Cd9CbDE883A267d5" }],
+    ["DNSRegistry", "antony.sh", { address: "0x2B5c7025998f88550Ef2fEce8bf87935f542C190" }],
+    ["CCIP Offchain RS", "luc.willbreak.eth", { address: "0x225f137127d9067788314bc7fcc1f36746a3c3B5" }],
+    ["CCIP Coinbase", "lucemans.cb.id", { address: "0x4e7abb71BEe38011c54c30D0130c0c71Da09222b" }],
+];
```
Comment (lucemans) [inline test/data/basic.ts:7]:
```suggestion
];

```

File: test/package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "name": "enstate-tests",
+  "module": "index.ts",
+  "type": "module",
+  "devDependencies": {
+    "bun-types": "latest"
+  },
+  "peerDependencies": {
+    "typescript": "^5.0.0"
+  }
+}
```
Comment (lucemans) [inline test/package.json:11]:
```suggestion
}

```

File: test/src/test_implementation.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    describe("t/" + function_name, () => {
+        for (const [label, input, partial] of dataset) {
+            test(label + ` (${input})`, async () => {
+                let output = await x(input);
+
+                expect(output).toMatchObject(partial);
+            });
+        }
+    });
+};
```
Comment (lucemans) [inline test/src/test_implementation.ts:14]:
```suggestion
};

```

Comment (lucemans) [inline test/src/test_implementation.ts:14]:
```suggestion
};

```

**Comment by Antony1060** [comment]
Needs more complete tests before merge.

**Review by Antony1060 — APPROVED** [review]
LGTM

**Pull request merged**